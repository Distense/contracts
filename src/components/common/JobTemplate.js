import React from 'react'
import PropTypes from 'prop-types';
import { Container, Header, List, Button } from 'semantic-ui-react'
import PageTitling from './PageTitling';

const JobTemplate = ({
  title,
  subtitle,
  description,
  idealCandidates,
  skills,
  salaryRange
}) => (
  <Container>
    <PageTitling title={title} subtitle={subtitle} />
    <hr />
    <p>{description}</p>
    <Header size="medium">
      Our ideal candidates:
    </Header>
    <p>{idealCandidates}</p>
    <Header size="medium">
      Youâ€™re familiar with several points from this list:
    </Header>
    <List bulleted items={skills} />
    <Header size="medium">
      Salary Range:
    </Header>
    <p>{salaryRange}</p>
    <a href="mailto:hiring@disten.se">
      <Button primary>
        Apply Now
      </Button>
    </a>
  </Container>
);

export default JobTemplate;

JobTemplate.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  idealCandidates: PropTypes.string.isRequired,
  skills: PropTypes.arrayOf(PropTypes.string).isRequired,
  salaryRange: PropTypes.string.isRequired,
}
