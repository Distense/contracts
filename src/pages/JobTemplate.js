import React, { Component } from 'react'
import { connect } from 'react-redux'

import { Menu, Grid, Header, Container, Button} from 'semantic-ui-react'

import PageTitling from '../components/common/PageTitling'
// import companyDescr_i_ption from './CompanyDesciption'

class JobTemplate extends Component {
  // shouldComponentUpdate(nextProps) {
  //   return this.props.ABCD !== nextProps.ABCD
  // }

  // componentWillMount() {
  // }

  render() {
    // const {  } = this.props

    return (
       <div>
       <PageTitling title="Associates(Specialized Developers)" subTitle="Blockchain Zoo"/>
       <Header 
          as="h3"
          classname="inconsolata"
          content="Company Desciption"
          style={{
                marginBottom: 0,
              }}
          >Company Desciption
          </Header>
          <p>
            Lorem ipsum ullamco laboris dolor enim tempor ut nostrud ex ullamco anim ad id duis ut.
            Lorem ipsum ullamco laboris dolor enim tempor ut nostrud ex ullamco anim ad id duis ut.
            Aute qui nostrud tempor tempor in aliquip magna sunt cupidatat adipisicing pariatur ea ut consectetur dolore exercitation culpa minim voluptate anim occaecat exercitation.
          </p>

       <Header 
          as="h3"
          classname="inconsolata"
          content="Ideal Candidate"
          style={{
                marginBottom: 0,
              }}
          >Ideal Candidate
          </Header>
          <p>
            Lorem ipsum ullamco laboris dolor enim tempor ut nostrud ex ullamco anim ad id duis ut.
            Lorem ipsum ullamco laboris dolor enim tempor ut nostrud ex ullamco anim ad id duis ut.
            Aute qui nostrud tempor tempor in aliquip magna sunt cupidatat adipisicing pariatur ea ut consectetur dolore exercitation culpa minim voluptate anim occaecat exercitation.
          </p>

       <Header 
          as="h3"
          classname="inconsolata"
          content="Desired Skills"
          style={{
                marginBottom: 0,
              }}
          >Desired Skills
          </Header>
          <ul>
          <li>UI/UX</li>
          <li>java</li>
          <li>javascript</li>
          <li>Nodejs</li>
          </ul>
      
       <Header 
          as="h3"
          classname="inconsolata"
          content="Salary Compensation"
          style={{
                marginBottom: 0,
              }}
        >Salary Compensation
        </Header>
        <p>10-100k</p>

       <Button primary>Email</Button>
      
       </div>

  )
 }
}

const mapStateToProps = state => ({
  // ABCD: getTotalSupplyDID(state)
})

const mapDispatchToProps = dispatch => ({
  // XYZ: () => dispatch(XYZ())
})

export default connect(mapStateToProps, mapDispatchToProps)(JobTemplate)
