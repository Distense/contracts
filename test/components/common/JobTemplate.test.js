import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import JobTemplate from '../../../src/components/common/JobTemplate';

function setup() {
  const props = {
    title: 'Im the title!',
    subtitle: 'Im the subtitle.',
    description: 'This is my description.',
    idealCandidates: 'Attributes for our ideal candidates.',
    skills: ['JavaScript', 'Ruby', 'Node.js', 'Solidity'],
    salaryRange: '100 - 110k',
  }

  const jobTemplate = shallow(<JobTemplate {...props} />);

  return { jobTemplate, props };
}

describe('<JobTemplate />', () => {
  it('renders without crashing', () => {
    const { jobTemplate } = setup();
    expect(jobTemplate).toBeDefined();
  });

  test('snapshot', () => {
    const { props } = setup();
    const mockstore = configureMockStore();
    const store = mockstore({ Locale: 'en' });
    const tree = renderer.create((
      <Provider store={store} >
        <MemoryRouter>
          <JobTemplate {...props} />
        </MemoryRouter>
      </Provider>
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
