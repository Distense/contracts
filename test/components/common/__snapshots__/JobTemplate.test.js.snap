// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JobTemplate /> snapshot 1`] = `
<div
  className="ui container"
>
  <div
    className="sixteen wide column"
  >
    <h2
      className="ui header"
    >
      Im the title!
    </h2>
    <p
      className="pagetitling-subtitle"
    >
      Im the subtitle.
    </p>
    <style
      jsx={true}
    >
      
        .pagetitling-subtitle {
          font-size: 1.3rem;
        }
      
    </style>
  </div>
  <hr />
  <p>
    This is my description.
  </p>
  <div
    className="ui medium header"
  >
    Our ideal candidates:
  </div>
  <p>
    Attributes for our ideal candidates.
  </p>
  <div
    className="ui medium header"
  >
    Youâ€™re familiar with several points from this list:
  </div>
  <div
    className="ui bulleted list"
    role="list"
  >
    <div
      className="item"
      data-value={undefined}
      onClick={[Function]}
      role="listitem"
    >
      JavaScript
    </div>
    <div
      className="item"
      data-value={undefined}
      onClick={[Function]}
      role="listitem"
    >
      Ruby
    </div>
    <div
      className="item"
      data-value={undefined}
      onClick={[Function]}
      role="listitem"
    >
      Node.js
    </div>
    <div
      className="item"
      data-value={undefined}
      onClick={[Function]}
      role="listitem"
    >
      Solidity
    </div>
  </div>
  <div
    className="ui medium header"
  >
    Salary Range:
  </div>
  <p>
    100 - 110k
  </p>
  <a
    href="mailto:hiring@disten.se"
  >
    <button
      className="ui primary button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      Apply Now
    </button>
  </a>
</div>
`;
